numba_cuda/numba/cuda/cudadrv/devicearray.py

```diff
@@ -901,8 +900,12 @@ def auto_device(obj, stream=0, copy=True, user_explicit=False):
     """
     if _driver.is_device_memory(obj):
         return obj, False
-    elif hasattr(obj, "__cuda_array_interface__"):
-        return numba.cuda.as_cuda_array(obj), False
```

Comment:
Before this PR, there were total of 3 (!) accesses of `__cuda_array_interface__` per call to `auto_device`.